src/main/java/com/github/caldav4j/CalDAVResource.java

```diff
@@ -16,47 +16,42 @@
  */
 package com.github.caldav4j;
 
-import com.github.caldav4j.model.response.CalendarDataProperty;
-import net.fortuna.ical4j.model.Calendar;
-import org.apache.jackrabbit.webdav.MultiStatusResponse;
-
 import java.io.Serializable;
 
 /**
  * A serializable class representing the Calendar along with the associated metadata. Used for
  * storing into the cache.
  */
-public class CalDAVResource implements Serializable{
+public class CalDAVResource<T extends Serializable> implements Serializable {
+    
 	private static final long serialVersionUID = -2607152240683030192L;
 	private ResourceMetadata resourceMetadata = null;
-	private Calendar calendar = null;
+	private T payload = null;
 
-	/**
-	 * Construct a Resource based on the Response.
-	 *
-	 * @param response Response to contruct from.
-	 */
-	public CalDAVResource(MultiStatusResponse response) {
```

Comment:
Q: Do we support multistatus responses after this patch?

Reply:
After checking the source, this constructor has been replaced with the ResponseToResource<T> class, which has this constructor, and thus should support MultiStatusResponses.